nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/DeleteHBaseRow.java

```diff
@@ -0,0 +1,183 @@
+
+import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+@WritesAttribute( attribute = "restart.index", description = "If a delete batch fails, it will restart from restart.index" )
+@Tags({ "delete", "hbase" })
+@CapabilityDescription(
+        "Delete HBase records individually or in batches. The input can be a single row ID in the body, one ID per line, " +
+        "row IDs separated by commas or a combination of the two. ")
+public class DeleteHBaseRow extends AbstractDeleteHBase {
+    static final AllowableValue ROW_ID_BODY = new AllowableValue("body", "FlowFile content", "Get the row key(s) from the flowfile content.");
+    static final AllowableValue ROW_ID_ATTR = new AllowableValue("attr", "FlowFile attributes", "Get the row key from an expression language statement.");
+
+    static final PropertyDescriptor ROW_ID_LOCATION = new PropertyDescriptor.Builder()
+            .name("delete-hb-row-id-location")
+            .displayName("Row ID Location")
+            .description("The location of the row ID to use for building the delete. Can be from the content or an expression language statement.")
+            .required(true)
+            .defaultValue(ROW_ID_BODY.getValue())
+            .allowableValues(ROW_ID_BODY, ROW_ID_ATTR)
+            .addValidator(Validator.VALID)
+            .build();
+
+    static final PropertyDescriptor FLOWFILE_FETCH_COUNT = new PropertyDescriptor.Builder()
+            .name("delete-hb-flowfile-fetch-count")
+            .displayName("Flowfile Fetch Count")
+            .description("The number of flowfiles to fetch per run.")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("5")
+            .expressionLanguageSupported(false)
+            .build();
+
+    static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("delete-hb-row-ff-count")
+            .displayName("Batch Size")
+            .description("The number of deletes to send per batch.")
+            .required(true)
+            .defaultValue("50")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(false)
+            .build();
+
+    static final PropertyDescriptor KEY_SEPARATOR = new PropertyDescriptor.Builder()
+            .name("delete-hb-separator")
+            .displayName("Delete Row Key Separator")
+            .description("The separator character(s) that separate multiple row keys " +
+                    "when multiple row keys are provided in the flowfile body")
+            .required(true)
+            .defaultValue(",")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+    static final PropertyDescriptor CHARSET = new PropertyDescriptor.Builder()
+            .name("Character Set")
```

Comment:
Please use `name` and `displayName` to align with other properties.

Reply:
Done